src/api/general-services/pipeline-manage/index.js

```diff
@@ -41,17 +44,21 @@ const getClusterInfo = async () => {
 };
 
 const constructPipelineStep = (context, step) => {
-  const { XStepType: stepType } = step;
+  const { XStepType: stepType, XConstructorArgs: args } = step;
 
   /* eslint-disable global-require */
   switch (stepType) {
     case 'delete-completed-jobs': {
       const f = require('./constructors/delete-complete-jobs');
-      return f(context, step);
+      return f(context, step, args);
     }
     case 'create-new-job-if-not-exist': {
       const f = require('./constructors/create-new-job-if-not-exist');
-      return f(context, step);
+      return f(context, step, args);
+    }
+    case 'create-new-step': {
+      const f = require('./constructors/create-new-step');
```

Comment:
I see that the `create-new-step` function is used in here actually -- better to import `createNewStep` and call `createNewStep(context, step, args)`, that will make the code easier to read. I would recommend doing this for the other two cases too